lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3375,42 +3375,20 @@ DynamicallyReplacedDeclRequest::evaluate(Evaluator &evaluator,
   return nullptr;
 }
 
-/// If the given type conforms to `Differentiable` in the given context, returns
-/// the `ProtocolConformanceRef`. Otherwise, returns an invalid
-/// `ProtocolConformanceRef`.
-///
-/// This helper verifies that the `TangentVector` type witness is valid, in case
-/// the conformance has not been fully checked and the type witness cannot be
-/// resolved.
-static ProtocolConformanceRef getDifferentiableConformance(Type type,
-                                                           DeclContext *DC) {
-  auto &ctx = type->getASTContext();
-  auto *differentiableProto =
-      ctx.getProtocol(KnownProtocolKind::Differentiable);
-  auto conf =
-      TypeChecker::conformsToProtocol(type, differentiableProto, DC);
-  if (!conf)
-    return ProtocolConformanceRef();
-  // Try to get the `TangentVector` type witness, in case the conformance has
-  // not been fully checked.
-  Type tanType = conf.getTypeWitnessByName(type, ctx.Id_TangentVector);
-  if (tanType.isNull() || tanType->hasError())
-    return ProtocolConformanceRef();
```

Comment:
@CodaFi: the removal of this code should address your concerns from https://github.com/apple/swift/pull/30936/files#r406541238:

> This mix of semantic and syntactico-semantic checks doesn't feel right. Is the idea that you're trying to check for a conformance to this protocol, or to a well-formed conformance of this protocol?

We directly use `TypeChecker::conformsToProtocol` now, which is syntactic. Semantic checks (requesting `TangentVector` type witnesses) are performed and diagnosed on-demand.